llvm/lib/Target/EraVM/EraVMInstrInfo.td

```diff
@@ -1248,3 +1259,95 @@ def : Pat<(EraVMTrap), (PANICr R0)>;
+
+// to stack
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, GR256:$rs1), stackaddr:$stack),
+          (ADDrrs_v GR256:$rs0, GR256:$rs1, stackaddr:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, imm16:$rs1), stackaddr:$stack),
+          (ADDirs_v imm16:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, (load_code memaddr:$rs1)), stackaddr:$stack),
+          (ADDcrs_v memop:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v (load_code memaddr:$rs0), GR256:$rs1), stackaddr:$stack),
+          (ADDcrs_v memop:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v (load_stack stackaddr:$rs0), GR256:$rs1), stackaddr:$stack),
+          (ADDsrs_v stackin:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMAdd_v GR256:$rs0, (load_stack stackaddr:$rs1)), stackaddr:$stack),
+          (ADDsrs_v stackin:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+
+// ============================ Overflow SUBs =====================================
+// to register
+def : Pat<(EraVMSub_v GR256:$rs0, GR256:$rs1), (SUBrrr_v GR256:$rs0, GR256:$rs1, 0)>;
+def : Pat<(EraVMSub_v imm16:$rs0, GR256:$rs1), (SUBirr_v imm16:$rs0, GR256:$rs1, 0)>;
+def : Pat<(EraVMSub_v (load_code memaddr:$rs0), GR256:$rs1),
+          (SUBcrr_v memop:$rs0, GR256:$rs1, 0)>;
+def : Pat<(EraVMSub_v (load_stack stackaddr:$rs0), GR256:$rs1),
+          (SUBsrr_v stackin:$rs0, GR256:$rs1, 0)>;
+
+def : Pat<(EraVMSub_v GR256:$rs0, imm16:$rs1),
+          (SUBxrr_v imm16:$rs1, GR256:$rs0, 0)>;
+def : Pat<(EraVMSub_v GR256:$rs0, (load_code memaddr:$rs1)),
+          (SUByrr_v memop:$rs1, GR256:$rs0, 0)>;
+def : Pat<(EraVMSub_v GR256:$rs0, (load_stack stackaddr:$rs1)),
+          (SUBzrr_v stackin:$rs1, GR256:$rs0, 0)>;
+
+// to stack
+def : Pat<(store_stack (EraVMSub_v GR256:$rs0, GR256:$rs1), stackaddr:$stack),
+          (SUBrrs_v GR256:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+
+def : Pat<(store_stack (EraVMSub_v imm16:$rs0, GR256:$rs1), stackaddr:$stack),
+          (SUBirs_v imm16:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMSub_v (load_code memaddr:$rs0), GR256:$rs1), stackaddr:$stack),
+          (SUBcrs_v memop:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMSub_v (load_stack stackaddr:$rs0), GR256:$rs1), stackaddr:$stack),
+          (SUBsrs_v stackin:$rs0, GR256:$rs1, stackin:$stack, 0)>;
+
+def : Pat<(store_stack (EraVMSub_v GR256:$rs0, imm16:$rs1), stackaddr:$stack),
+          (SUBxrs_v imm16:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMSub_v GR256:$rs0, (load_code memaddr:$rs1)), stackaddr:$stack),
+          (SUByrs_v memop:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+def : Pat<(store_stack (EraVMSub_v GR256:$rs0, (load_stack stackaddr:$rs1)), stackaddr:$stack),
+          (SUBzrs_v stackin:$rs1, GR256:$rs0, stackin:$stack, 0)>;
+
+// ============================ Overflow ADDs =====================================
```

Comment:
Typo, it should be Overflow `MULs` and patterns for large_imm are missing.

Reply:
Correct it now.